src/middleware/github-server-app-middleware.ts

```diff
@@ -0,0 +1,31 @@
+import { NextFunction, Request, Response } from "express";
+import { GitHubServerApp } from "models/github-server-app";
+import { Installation } from "models/installation";
+
+export const githubServerAppMiddleware = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+	const { jiraHost } = res.locals;
+	const { id } = req.params;
+
+	if (id) {
+		req.log.info(`Retrieving GitHub app with id ${id}`);
+		const gitHubServerApp = await GitHubServerApp.getForGitHubServerAppId(Number(id));
+
+		if (!gitHubServerApp) {
+			req.log.error({ id, jiraHost }, "No GitHub app found for provided id.");
+			throw new Error("No GitHub app found for provided id.");
+		}
+
+		const installation = await Installation.getForHost(jiraHost);
+
+		if (installation?.id !== gitHubServerApp?.installationId) {
```

Comment:
I'd probably get the installation from the `gitHubServerApp.installationId` then compare the jiraHost instead :)

Reply:
You also don't need the `gitHubServerApp?` since you did the conditional check above